src/main/java/org/breedinginsight/services/brapi/BrAPIClientProvider.java

```diff
@@ -45,4 +48,18 @@ public BrAPIClient getClient(BrAPIClientType clientType){
         else { return genoClient; }
     }
 
+    public List<BrAPIClient> getAllUniqueClients(){
+
+        List<BrAPIClient> clients = new ArrayList<>();
+        clients.add(coreClient);
+        if (!coreClient.brapiURI().equals(phenoClient.brapiURI())){
+            clients.add(phenoClient);
+        }
+        if (!coreClient.brapiURI().equals(genoClient.brapiURI()) && !phenoClient.brapiURI().equals(genoClient.brapiURI())) {
+            clients.add(genoClient);
+        }
```

Comment:
This could be simplified by having the return type be a `Set` (and use the `TreeSet` implementation of `Set`):

```java
Set<BrAPIClient> clients = new TreeSet<>(Comparator.comparing(BrAPIClient::brapiURI));
```

Then just add the `core`, `pheno`, and `geno` clients

Reply:
Ah cool, I was wondering if you could customize the set comparator. I'll change this.

Reply:
Changed and pushed